torch_xla/csrc/ops/dynamic_ir.h

```diff
@@ -74,4 +75,18 @@ class SizeDiv : public XlaNode, public torch::lazy::DimensionNode {
   virtual XlaOpVector Lower(LoweringContext* loctx) const override;
 };
 
+class SizeConstant : public torch_xla::Scalar,
+                     public torch::lazy::DimensionNode {
+ public:
+  SizeConstant(int64_t val);
+  int64_t getStaticValue() const override { return value().to<int64_t>(); };
+  bool isSymbolic() const override { return false; };
+  std::string ToString() const override {
+    this->torch_xla::Scalar::ToString();
+  };
+  virtual XlaOpVector Lower(LoweringContext* loctx) const override {
+    TORCH_CHECK(false, "NYI");
```

Comment:
@JackCaoG 
I don't think we necessarily need to implement it if we aren't going to use XLA to evaluate size expressions? If we do, I can just call `Scalar::Lower` I believe?

Reply:
I think we would need lower and yes we can use the `Scalar::Lower`.

When we do `SizeAddNode` = `SizeConstant` + `SizeNode`, we want `SizeAddNode->lower` gives the correct result.

Reply:
got it! so you are planning to use XLA to evaluate those expressions?

Reply:
Yea, I think we have to otherwise we violoate the contract of the Node class. For every node, we expect after the lowered HLO represents its real value.

It is kind of awkward that we say a `SizeAddNode`'s lowering does not always reflect its real size.

Reply:
I am not sure if we are planning to support

```
t1 = torch.tensor(1)
t2 = some_dynamic_tensor
t1 *= (t2.size(0) + 5)
```

in which case the result of `SizeAddNode` in last line needs to be correct.